std/variant.d

```diff
@@ -2049,8 +2068,8 @@ unittest
     Algebraic!(int, float, string) variant2 = 5.0f;
     // Shouldn' t compile as float not handled by visitor.
     static assert(!__traits(compiles, variant2.visit!(
-                        (int) {},
-                        (string) {})()));
+                        (int i) {},
```

Comment:
Turns out it wasn't a bug (probably), it was just unclear. `(string){}` is **not** a function taking one unnamed `string` arg, but rather a function taking a single generic arg named "string". So this is more along the lines of making this test actually verify what it think it is (`compiles` was swallowing the failure, but it wasn't the expected failure).